lib/archive/stripper_test.go

```diff
@@ -45,36 +46,33 @@ func generateArchive(name string, num, size int) ([]string, io.Reader) {
 	// our file contents are zeros - this is the worst case for stripping trailing headers
 	zeros := make([]byte, size)
 
-	go func() {
-		for i := 0; i < num; i++ {
-			// we only really care about two things in the header
-			hdr := &tar.Header{
-				Name: index[i],
-				Size: int64(size),
-			}
-
-			fmt.Printf("Writing header for file %s:%d\n", name, i)
-			err := tw.WriteHeader(hdr)
-			if err != nil {
-				panic(err)
-			}
-
-			fmt.Printf("Writing data for file %s:%d\n", name, i)
-			n, err := tw.Write(zeros)
-			if err != nil {
-				panic(err)
-			}
-			if n != size {
-				panic(fmt.Sprintf("Failed to write all bytes: %d != %d", n, size))
-			}
+	for i := 0; i < num; i++ {
```

Comment:
Why are we removing the goroutine? The goroutine doesn't have that much of an effect in the test, but it's a better representation of how it gets used in reality.